src/main/java/de/tum/in/www1/artemis/service/exam/StudentExamService.java

```diff
@@ -529,4 +529,20 @@ public StudentExam deleteTestRun(Long testRunId) {
         studentExamRepository.deleteById(testRunId);
         return testRun;
     }
+
+    /**
+     * Extracts the exercise ids from the given studentExam
+     * @param studentExam which exercise ids should be extracted
+     * @return Long[] of the extracted exercise ids
+     */
+    public Long[] extractStudentExamExerciseIds(StudentExam studentExam) {
+        Exercise[] exercises = studentExam.getExercises().toArray(new Exercise[0]);
```

Comment:
instead of that you could use stream. I assume it will take at most two lines 

you need something like that: 
`List<Long> exerciseIds = studentExam.getExercises().stream.map(Exercise::getId).collect(toList())`

Reply:
Yes, you are correct it will take less space. I am just curious, what about the performance, if I am not mistaken using stream is more performant than just using a for loops if you have a big dataset, but for student exams simple for loops should be better? 😅 (I found this [article](https://blog.jdriven.com/2019/10/loop/#:~:text=There%20are%20many%20opinions%20about,parallel%20stream%20will%20perform%20better.&text=So%20although%20the%20difference%20is,loops%20win%20by%20pure%20performance.))

It works if I do this: `return studentExam.getExercises().stream().map(Exercise::getId).toArray(Long[]::new);`

Reply:
well, if it is small list go ahead with for-loop. guess I am just used to seeing stream instead of for 😄